tt_metal/impl/device/device.cpp

```diff
@@ -1647,6 +1836,12 @@ bool Device::initialize(const uint8_t num_hw_cqs, size_t l1_small_size, size_t t
 }
 
 bool Device::close() {
```

Comment:
Why do we have test code in actual working code?
Are we ever going to skip tear down?

Reply:
Unlikely to ever skip teardown, but `Device::close()` is automatically called on device pool destruction, so wasn't sure how to force a teardown skip without putting it here or in the device pool

Reply:
In the test can we do a TT_ASSERT or TT_FATAL to cause the test to end without device close?
That would model an abrupt test end more accurately rather than just skipping stpes in

Reply:
Updated this in latest commit